docs/API_DESIGN.md

```diff
@@ -90,11 +92,29 @@ CLASS VerifiableCredential
   PUBLIC DATA credentialSubject: CredentialSubject
 
   CONSTRUCTOR create(issuer: Issuer, credential_subject: CredentialSubject, options: CreateOptions)
+  CONSTRUCTOR decode(vc_jwt: string)
```

Comment:
I remember we went though this before with the decode vs verify vs both on the js side. 

Basically definitely don't have both as it is confusing to the end user. 

I would do the superset of functionally, I would have `verify` and `verifyOptions`.

 But now is always the question:
 
 of if it's expired or something and your verify just throws or returns a bool, how would the user know what is in there if they decided to decode it? 
 
 Could implement a verify results object, but this also causes confusion

Reply:
btw in your current impl I think you wanted VerifiableCredentialDecodeOptions on the decode too right?

Reply:
I'm reading this wrong, this is a new constructor not VerifiableCredential.decode() ok let me rethink lol

Reply:
another option is to have `create_from_vc_jwt(vc_jwt, createFromVcJwtOptions)`

Reply:
> btw in your current impl I think you wanted VerifiableCredentialDecodeOptions on the decode too right?

yes! fixed 👍

Reply:
> Basically definitely don't have both as it is confusing to the end user

This was my driving principal as I was thinking it through as well. However, I will add [that `web5-go` does actually offer both](https://github.com/TBD54566975/web5-go/blob/main/vc/vcjwt.go), so I'm not strictly opposed to it.

If we only have one entrypoint, then it must be named either `verify()` or `decode()`, right? I went with `decode()` with the option `verify: bool` parameter because both features are expected to decode the VC-JWT, but only 1 or the 2 will execute cryptographic verification (& also, @nitro-neal would you bucket semantic data model validation under that same `verify` feature or should we distinguish between validation & verification? So that would be two options `verify: bool` and `validate: bool`?).

Reply:
yea! we went down that road too lol validate vs verify.

I think we should just have verify and it should validate under the hood.

Reply:
Actually I think this approach is slightly different because decode is a constructor where as verify could be a method on the object:

```
  CONSTRUCTOR create(issuer: Issuer, credential_subject: CredentialSubject, options: CreateOptions)
  CONSTRUCTOR decode(vc_jwt: string, options: VerifiableCredentialDecodeOptions) // you almost dont need options here, for more clarity this could be called from_vc_jwt or something too.., and then in kotlin or something we could pug sugar around it where we have 2 constructors, one for issuer, cred, and options, another for vcjwt
  METHOD sign(options: VerifiableCredentialSignOptions)
  METHOD verify(options: VerifiableCredentialVerifyOptions) // this validates too
  ```

Reply:
Okay I like that, and yeah I would propose we go with `from_vc_jwt()` to be consistent with our `from_json_string()`. Then that would raise the question, should we name `sign()` to `to_vc_jwt()` 🤔 

The reason I don't _love_ the design you lay out, is there is a code path from `create()` to `verify()` which is non-sensical because `verify()` is only relevant if the originating information was a VC-JWT. Or, put more concretely, that `verify()` has a signature to verify against. Any thoughts on this? Maybe we just handle that with an error, or maybe if the VC-JWT isn't known to the instance then it only performs data model validation and simply ignores cryptographic verification?